features/discovery/src/main/java/org/opennms/netmgt/discovery/messages/DiscoveryJob.java

```diff
@@ -98,4 +101,25 @@ public String toString() {
                  .add("ranges", m_ranges)
                  .toString();
     }
+    
+    /**
+     * <P>
+     * Returns int : the total task timeout for all ip ranges
+     * </P>
+     */
+    public int calculateTaskTimeout(){
```

Comment:
This method is very hard to read. Can you reformat it into a sequence of BigInteger calls instead of using 5 different variables? ie:
``` 
BigInteger value = BigInteger.valueOf(range.getRetries())
.multiply(BigInteger.valueOf(range.getAddressRange().getSizeOfIpAddrRange()))
.multiply(...)
```